x-pack/plugins/canvas/public/lib/element.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import defaultHeader from './default_header.png';
+
+export class Element {
```

Comment:
It's used [here](https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/lib/elements_registry.js#L12).  Since the individual elements actually return plain string/any objects, they require a type that is not linked to the underlying `Element` class.  This is how I chose to approach it, but I'm open to a different standard, (if one exists).

Reply:
> Since the individual elements actually return plain string/any objects,

When you say "the individual elements", do you mean these "CanvasElement" functions? Or some function on the `Element` class? 
 
Up to you, but I think the `Element` and `CanvasElement` naming scheme is a tad confusing.  

When I read the two names, my first instinct is to think that `CanvasElement` is a specific implementation of the more generic `Element`.  Which maybe it kind of is, but given that one is a function returning a config,  and one is a class throws it off for me (I'd instead be looking for `class CanvasElement extends Element` somewhere.  

What about changing CanvasElement to `CanvasElementConfigFactory`?

@w33ble - while Clint and I were chatting, I noticed that all these configuration factory functions could, (i think), theoretically just return the const and not be a function at all.

e.g. instead of:
```
export const verticalProgressBar: CanvasElement = () => ({
  name: 'verticalProgressBar',	  name: 'verticalProgressBar',
  displayName: 'Vertical progress bar',	  displayName: 'Vertical progress bar',
  help: '...',
   expression: `...`,
 });
```

just do:

```
export const verticalProgressBar: CanvasElementConfig = {
  name: 'verticalProgressBar',	  name: 'verticalProgressBar',
  displayName: 'Vertical progress bar',	  displayName: 'Vertical progress bar',
  help: '...',
   expression: `...`,
 };
```

But there is probably a reason you and/or @rashidkpc made it a function. Just curious as to the background on that.  Perhaps you were thinking in the future of passing in variables?

Reply:
> But there is probably a reason you and/or @rashidkpc made it a function

Yeah, it had to do with when the code was executed on load. Some parts of the system needed to be functions to defer when the code was run; functions allow us to be explicit about it. We changed everything to match, even when it wasn't a requirement, so that the plugin API was consistent.